datadog-sidecar/src/service/telemetry.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::{HashMap, HashSet};
+use std::ops::Sub;
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::OnceLock;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, Instant, SystemTime};
+
+use ddcommon::{tag::Tag, MutexExt};
+use ddtelemetry::data::{self, Integration};
+use ddtelemetry::metrics::{ContextKey, MetricContext};
+use ddtelemetry::worker::{TelemetryActions, TelemetryWorkerHandle};
+use manual_future::ManualFuture;
+use serde::Deserialize;
+use serde_with::{serde_as, VecSkipError};
+use tokio::time::sleep;
+use tracing::warn;
+
+use crate::service::SidecarAction;
+
+//
+// ──────────────────────────────────────────────
+//   ⬇ Type aliases and statics
+// ──────────────────────────────────────────────
+//
+
+type ComposerCache = HashMap<PathBuf, (SystemTime, Arc<Vec<data::Dependency>>)>;
+static COMPOSER_CACHE: OnceLock<tokio::sync::Mutex<ComposerCache>> = OnceLock::new();
+static LAST_CACHE_CLEAN: OnceLock<AtomicU64> = OnceLock::new();
+
+fn get_composer_cache() -> &'static tokio::sync::Mutex<ComposerCache> {
+    COMPOSER_CACHE.get_or_init(|| tokio::sync::Mutex::new(HashMap::new()))
+}
+
+fn get_last_cache_clean() -> &'static AtomicU64 {
+    LAST_CACHE_CLEAN.get_or_init(|| AtomicU64::new(0))
+}
```

Comment:
We recently replaced OnceLock occurences by LazyLock in #1096. Can you also use LazyLock here? (You'll have to rebase on main first.)